cmake/EthUtils.cmake

```diff
@@ -62,3 +62,13 @@ macro(eth_add_test NAME)
 
 endmacro()
 
+# Creates C resources file from files
+function(eth_add_resources TARGET RESOURCE_FILE)
+
+	add_custom_target("${TARGET}.resources"
```

Comment:
You don't need PRE_BUILD here, see how it is done for rcc:
```cmake```
        add_custom_command(OUTPUT ${outfile}
                           COMMAND ${Qt5Core_RCC_EXECUTABLE}
                           ARGS ${rcc_options} -name ${outfilename} -o ${outfile} ${infile}
                           MAIN_DEPENDENCY ${infile}
                           DEPENDS ${_RC_DEPENDS} "${out_depends}" VERBATIM)
```
You just need to add ${outfile} to source list after that